packages/espresso/android/src/main/java/androidx/test/espresso/flutter/internal/protocol/impl/DartVmServiceUtil.java

```diff
@@ -71,25 +70,15 @@ public static String getDartIsolateId(View flutterView) {
   }
 
   /** Gets the Dart executor for the given {@code flutterView}. */
-  @SuppressWarnings("deprecation")
-  public static DartExecutor getDartExecutor(View flutterView) {
-    checkNotNull(flutterView, "The Flutter View instance cannot be null.");
-    // Flutter's embedding is in the phase of rewriting/refactoring. Let's be compatible with both
-    // the old and the new FlutterView classes.
-    if (flutterView instanceof io.flutter.view.FlutterView) {
-      return ((io.flutter.view.FlutterView) flutterView).getDartExecutor();
-    } else if (flutterView instanceof io.flutter.embedding.android.FlutterView) {
-      FlutterEngine flutterEngine =
-          ((io.flutter.embedding.android.FlutterView) flutterView).getAttachedFlutterEngine();
-      if (flutterEngine == null) {
-        throw new FlutterProtocolException(
-            String.format(
-                "No Flutter engine attached to the Flutter view [id: %d].", flutterView.getId()));
-      }
-      return flutterEngine.getDartExecutor();
-    } else {
+  public static DartExecutor getDartExecutor(@Nonnull FlutterView flutterView) {
+    // TODO(gmackall): getAttachedFlutterEngine() is marked as @visibleForTesting - determine if
```

Comment:
Not a fan of Visible for testing methods being used in production code.

Reply:
Agreed, but I suppose given that this is the `espresso` plugin this is indeed actually used for testing?

Reply:
Yeah you are correct this is test code.